inc/datahandlers/user.php

```diff
@@ -1783,4 +1783,61 @@ function clear_profile($delete_uids=false, $gid=0)
 			remove_avatars($uid);
 		}
 	}
+
+	public function verify_signature()
+	{
+		global $mybb, $parser;
+
+		$parser_options = array(
+			'allow_html' => $mybb->settings['sightml'],
+			'filter_badwords' => 1,
+			'allow_mycode' => $mybb->settings['sigmycode'],
+			'allow_smilies' => $mybb->settings['sigsmilies'],
+			'allow_imgcode' => $mybb->settings['sigimgcode'],
+			"filter_badwords" => 1
+		);
+
+		if($mybb->user['showimages'] != 1 && $mybb->user['uid'] != 0)
+		{
+			$parser_options['allow_imgcode'] = 0;
+		}
+
+		$parsed_sig = $parser->parse_message($mybb->get_input('signature'), $parser_options);
```

Comment:
Same here, the only difference is that you pass signature as data this time, but aren't using the data.